gazebo_plugins/src/gazebo_ros_wheel_slip.cpp

```diff
@@ -87,39 +103,54 @@ void GazeboRosWheelSlip::Load(gazebo::physics::ModelPtr _model, sdf::ElementPtr
       wheel_element = wheel_element->GetNextElement("wheel"))
     {
       auto wheel_name = wheel_element->Get<std::string>("link_name");
-      double slip_lateral = wheel_element->Get<double>("slip_compliance_lateral");
-
-      if (slip_lateral >= 0.) {
-        this->impl_->default_slip_lateral_ = slip_lateral;
-        this->impl_->map_slip_lateral_default_[wheel_name] = slip_lateral;
-      } else {
-        this->impl_->default_slip_lateral_ = 0.0;
-        this->impl_->map_slip_lateral_default_[wheel_name] = 0.0;
-        auto msg = "Negative slip lateral compliance value found in sdf for " + wheel_name +
-          " will be set to 0.0";
-        RCLCPP_INFO(
-          this->impl_->ros_node_->get_logger(),
-          msg.c_str()
-        );
-      }
 
-      double slip_longitudinal = wheel_element->Get<double>("slip_compliance_longitudinal");
-
-      if (slip_longitudinal >= 0.) {
-        this->impl_->default_slip_longitudinal_ = slip_longitudinal;
-        this->impl_->map_slip_longitudinal_default_[wheel_name] = slip_longitudinal;
-      } else {
-        this->impl_->default_slip_longitudinal_ = 0.0;
-        this->impl_->map_slip_longitudinal_default_[wheel_name] = 0.0;
-        auto msg = "Negative slip longitudinal compliance value found in sdf for " + wheel_name +
-          " will be set to 0.0";
-        RCLCPP_INFO(
-          this->impl_->ros_node_->get_logger(),
-          msg.c_str()
-        );
-      }
+      const double slip_lateral = this->impl_->validate_wheel_slip_value(
+        wheel_element->Get<double>("slip_compliance_lateral"),
+        "lateral",
+        wheel_name);
+      this->impl_->map_slip_lateral_default_[wheel_name] = slip_lateral;
+
+      const double slip_longitudinal = this->impl_->validate_wheel_slip_value(
+        wheel_element->Get<double>("slip_compliance_longitudinal"),
+        "longitudinal",
+        wheel_name);
+      this->impl_->map_slip_longitudinal_default_[wheel_name] = slip_longitudinal;
+    }
+  }
+
+  // Check for "global" slip parameters and use them for individual wheels if provided
+  impl_->ros_node_->declare_parameter(
+    "slip_compliance_unitless_lateral", rclcpp::PARAMETER_DOUBLE);
+  impl_->ros_node_->declare_parameter(
+    "slip_compliance_unitless_longitudinal", rclcpp::PARAMETER_DOUBLE);
```

Comment:
Does this mean the parameter ``slip_compliance_unitless_longitudinal`` will have no default value ? Will the parameters return 0 if we do a "get value" on these parameters ?

Reply:
In this PR : https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1308, we had explicit set their default values due to some weird behavior.

Reply:
The parameters will return "not set", which I think is reasonable if the user did not supply any values. 

I don't recall what the undesired behavior was in #1308, but this PR is certainly changing behavior so maybe the concern no longer applies?